unblob/finder.py

```diff
@@ -139,11 +139,15 @@ def search_chunks(  # noqa: C901
     scanner = hyperscan_db.build(hyperscan_context, _hyperscan_match)
 
     try:
-        if scanner.scan(file) == Scan.Terminate:
-            logger.debug(
-                "Scanning terminated as chunk matches till end of file",
-            )
-            return all_chunks
+        for i in range(0, file.size(), DEFAULT_BUFSIZE):
+            if (
+                scanner.scan(file[i : i + DEFAULT_BUFSIZE])  # noqa: E203
+                == Scan.Terminate
+            ):
+                logger.debug(
+                    "Scanning terminated as chunk matches till end of file",
+                )
+                return all_chunks
```

Comment:
When removing the logging (which was already logged on line 111 above) and instead of returning immediately just exiting the loop like this:
```suggestion
        for i in range(0, file.size(), DEFAULT_BUFSIZE):
            if (
                scanner.scan(file[i : i + DEFAULT_BUFSIZE])  # noqa: E203
                == Scan.Terminate
            ):
                break
```

We ended up with a non-trivial code chunk, that has been repeated twice more in other files (`xz` and `bzip2` handlers).

I propose to extract this common functionality to a separate function in `file_utils`:

```py
def stream_scan(scanner, file: File):
    for i in range(0, file.size(), DEFAULT_BUFSIZE):
        if (
            scanner.scan(file[i : i + DEFAULT_BUFSIZE])  # noqa: E203
            == Scan.Terminate
        ):
            break
```

(Maybe also add a type annotation for `scanner` and a doc-string)